modules/nextflow/src/main/groovy/nextflow/trace/TraceObserver.groovy

```diff
@@ -120,18 +122,17 @@ interface TraceObserver {
      * @param trace
      *      The associated {@link TraceRecord} for the current task.
      */
-    default void onFlowError(TaskHandler handler, TraceRecord trace){}
+    void onFlowError(TaskHandler handler, TraceRecord trace){}
 
     /**
-     * Method that is invoked when a workflow output is published.
+     * Method that is invoked when a value is published from a channel.
+     *
+     * NOTE: This method is no longer used.
      *
-     * @param name
-     *      The name of the workflow output
      * @param value
-     *      A list if the published channel was a queue channel,
-     *      otherwise an object if the channel was a value channel
      */
-    default void onWorkflowPublish(String name, Object value){}
+    @Deprecated
```

Comment:
If all the class is deprecated, is it still required?

Reply:
Yes, because almost all plugins still use it. We need to migrate them and give time to third party to do the same